datafusion/sql/src/planner.rs

```diff
@@ -5348,6 +5363,27 @@ mod tests {
         quick_test(sql, expected);
     }
 
+    #[test]
+    fn udaf_as_window_func() -> Result<()> {
+        let my_max = create_udaf(
+            "my_max",
+            DataType::Int32,
+            Arc::new(DataType::Int32),
+            Volatility::Immutable,
+            Arc::new(|_| Ok(Box::new(MaxAccumulator::try_new(&DataType::Int32)?))),
```

Comment:
I wonder how much work / copying it would be to avoid the test dependency on `datafusion-physical-expr`. It doesn't seem like we actually need anything about the `MaxAccumulator` in this code

Reply:
I've created a new udaf in testing codes and get rid of `datafusion-physical-expr` in dev-depencency. However, to write udaf (and other tests), I have to dependent on `datafusion-core`.

Reply:
I have a follow on PR https://github.com/apache/arrow-datafusion/pull/4589 I will propose after this one is merged to move the test (and thus avoid this dependency). I would like to merge this PR as is so it doesn't accumulate conflicts